task2/src/router/userRouter.ts

```diff
@@ -0,0 +1,75 @@
+import express, { Request, Response } from 'express';
+import { getAutoSuggestUsers } from '../utils/getAutoSuggestUsers';
+import { UserRequestSchema, validator } from '../middleware/validator';
+import { UserValidationSchema } from '../schemas/User';
+import { ValidatedRequest } from 'express-joi-validation';
+import { User } from '../types/User';
+import { v4 as uuid } from 'uuid';
+
+export let users: User[] = [
+    { age: 23, isDeleted: false, login: 'Dmytro', password: 'admin', id: uuid() },
+    { age: 24, isDeleted: false, login: 'Pavlo', password: 'mod', id: uuid() },
+    { age: 27, isDeleted: false, login: 'Oleh', password: 'guest', id: uuid() },
+    { age: 18, isDeleted: false, login: 'Zekora', password: 'guest', id: uuid() },
+    { age: 44, isDeleted: false, login: 'Artem', password: 'guest', id: uuid() },
+    { age: 20, isDeleted: false, login: 'Daniel', password: 'mod', id: uuid() }
+];
+
+export const userRouter = express.Router();
+
+userRouter.param('id', (req: Request, res: Response, next, id) => {
+    const userById = users.find((user) => user.id === id);
+    if (userById) req.user = userById;
+    next();
+});
+
+userRouter.route('/users')
+    .get((req: Request, res: Response) => {
+        let { limit, filter } = req.query;
+        if (!limit) limit = '10';
+        if (!filter) filter = '';
```

Comment:
Default parameters in this exact case won't help, because if we remove **ifs**, the value will always be passed anyway, and the default parameters will never be triggered

e.g.: if **limit** or **filter** is missing **String()** will return 'undefined' and go deep into **getAutoSuggestUsers** method, hence default param won't work.

But anyway I understand that my solution with **ifs** not correct

Reply:
Instead IF's created separate middleware for handling default query parameters. (Hardcoded value there)